metadata-db-app/src/cmr/metadata_db/data/oracle/search.clj

```diff
@@ -142,25 +149,51 @@
   "Retrieve concept maps from the given table, handling small providers separately from
   normal providers."
   (fn [db table concept-type providers params]
-    (or (:small (first providers))
-        (= :variable concept-type)
-        (= :service concept-type)
-        (= :tool concept-type)
-        (= :subscription concept-type))))
+    (case concept-type
+      :subscription :subscription
+      (or (:small (first providers))
+          (= :variable concept-type)
+          (= :service concept-type)
+          (= :tool concept-type)))))
 
 ;; Execute a query against a single table where provider_id is a column
 (defmethod find-concepts-in-table true
   [db table concept-type providers params]
   (let [fields (columns-for-find-concept concept-type params)
-        params (params->sql-params concept-type
-                                   providers
-                                   (assoc params :provider-id (map :provider-id providers)))
-        stmt (gen-find-concepts-in-table-sql concept-type table fields params)]
+        sql-params (params->sql-params concept-type
+                                       providers
+                                       (assoc params :provider-id (map :provider-id providers)))
+        stmt (gen-find-concepts-in-table-sql concept-type table fields sql-params)]
+    (j/with-db-transaction
+      [conn db]
+      (doall
+        (mapv #(oc/db-result->concept-map concept-type conn (:provider_id %) %)
+              (su/query conn stmt))))))
+
+(defn- add-subscription-last-notified-at
+  [db subscription]
+  (let [concept-id (:concept-id subscription)]
+    (if (notifications/sub-notification-exists? db concept-id)
+      (assoc subscription :last-notified-at (:last-notified-at (notifications/get-sub-notification db concept-id)))
+      subscription)))
+
+(defmethod find-concepts-in-table :subscription
+  [db table concept-type providers params]
+  (let [fields (columns-for-find-concept concept-type (dissoc params :include_last_notified_at))
+        sql-params (params->sql-params concept-type
+                                       providers
+                                       (assoc params :provider-id (map :provider-id providers)))
+        stmt (gen-find-concepts-in-table-sql concept-type table fields sql-params)]
     (j/with-db-transaction
-     [conn db]
-     (doall
-      (mapv #(oc/db-result->concept-map concept-type conn (:provider_id %) %)
-            (su/query conn stmt))))))
+      [conn db]
+      (doall
+        (mapv
+          #(if (:include_last_notified_at params)
```

Comment:
I think we have talked about it and we want to query the database with one query and getting all fields back, not making multiple sql queries to get the last notified time. This code actually making two calls for each subscription concept to get the last notification time.